src/server/pfs/server/driver.go

```diff
@@ -698,21 +715,19 @@ func (d *driver) finishCommit(ctx context.Context, commit *pfs.Commit, tree *pfs
 		commitInfo.Description = description
 	}
 
-	prefix, err := d.scratchCommitPrefix(ctx, commit)
-	if err != nil {
-		return err
-	}
+	scratchPrefix := d.scratchCommitPrefix(commit)
 	defer func() {
-		// Delete the scratch space for this commit
-		_, err = d.etcdClient.Delete(ctx, prefix, etcd.WithPrefix())
-		if err != nil && retErr == nil {
-			retErr = err
+		if retErr != nil {
+			return
 		}
```

Comment:
This is a nice semantic change here, I guess previously there were ways that `FinishCommit` could fail that would cause the scratch space to be cleaned up so you'd have to recreate the entire commit.

Reply:
Done